### ISSUE TITLE
A mismatch between best hyperparameters and hyperparameters ranges

### ISSUE THREAD

**Issue by kefirski** [QingruZhang/AdaLoRA#11]
Hello. We were trying to develop a new method on top of AdaLoRA and are faced with an inconvenience regarding the hyperparameter search used for GLUE. 
[](url)
With Appendix Section C, you have provided hyperparameter ranges used during hyperparameter search, and then you report best-performing learning rates. Although, the best-performing values that you have reported are not included in the search range. E.g., for the STS-B dataset, you [use](https://github.com/QingruZhang/AdaLoRA/blob/main/NLU/scripts/run_debertav3_stsb.sh) a learning rate equal to `2.2e-3`, which is not listed in Section C.2.

Could you help me understand what I am missing and which ranges I should use to ensure the reproducibility of your hyperparameter search?

![2023-08-14 09 42 55](https://github.com/QingruZhang/AdaLoRA/assets/8404265/388faf1d-2d97-4619-8ff3-00e624864f91)

**Comment by kefirski** [comment]
Furthermore, which ranks/target ranks are used to obtain results from Table 1? 

E.g., For AdaLoRA with 0.32M of trainable parameters, according to Appendix C.1, I should use `r=4` and `target_rank=2`. However, scripts provided with this repository contain various ranks beyond these values. 

For [CoLA](https://github.com/QingruZhang/AdaLoRA/blob/main/NLU/scripts/run_debertav3_cola.sh) and [MNLI](https://github.com/QingruZhang/AdaLoRA/blob/main/NLU/scripts/run_debertav3_mnli.sh) datasets `r=3` and `target_rank=1`
For [others](https://github.com/QingruZhang/AdaLoRA/blob/main/NLU/scripts/run_debertav3_stsb.sh), `r=2` and `target_rank=1`.

Am I correctly understanding that to reproduce results from Table 1 with 0.32M parameters, I should strictly use `r=4` and `target_rank=2`?

![Screenshot 2023-08-14 at 09 53 18](https://github.com/QingruZhang/AdaLoRA/assets/8404265/f7944012-f528-4c56-8b2e-0a214c1f9a94)

![Screenshot 2023-08-14 at 09 53 33](https://github.com/QingruZhang/AdaLoRA/assets/8404265/cfee6352-dd80-4c57-9f25-0565b5c6ae62)

**Comment by Heyest** [comment]
@kefirski Hello, I've encountered the same issue. I also tried setting r=2 and target_rank=1 on the MRPC dataset, but I found that the performance is not as good as the original settings in the script. Have you faced a similar issue, and how did you resolve it? 

> Furthermore, which ranks/target ranks are used to obtain results from Table 1?
> 
> E.g., For AdaLoRA with 0.32M of trainable parameters, according to Appendix C.1, I should use `r=4` and `target_rank=2`. However, scripts provided with this repository contain various ranks beyond these values.
> 
> For [CoLA](https://github.com/QingruZhang/AdaLoRA/blob/main/NLU/scripts/run_debertav3_cola.sh) and [MNLI](https://github.com/QingruZhang/AdaLoRA/blob/main/NLU/scripts/run_debertav3_mnli.sh) datasets `r=3` and `target_rank=1` For [others](https://github.com/QingruZhang/AdaLoRA/blob/main/NLU/scripts/run_debertav3_stsb.sh), `r=2` and `target_rank=1`.
> 
> Am I correctly understanding that to reproduce results from Table 1 with 0.32M parameters, I should strictly use `r=4` and `target_rank=2`?
> 
> ![Screenshot 2023-08-14 at 09 53 18](https://private-user-images.githubusercontent.com/8404265/260379384-f7944012-f528-4c56-8b2e-0a214c1f9a94.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE5NDk2NjYsIm5iZiI6MTcwMTk0OTM2NiwicGF0aCI6Ii84NDA0MjY1LzI2MDM3OTM4NC1mNzk0NDAxMi1mNTI4LTRjNTYtOGIyZS0wYTIxNGMxZjlhOTQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMDdUMTE0MjQ2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Nzg4ZWYxMTY3OTQzNmM1ZDJlZWFjNTNhZDg3ZGFjODBmZjM5ZDkyOGE5Yzg2YjQ3YzVhNzQxMmJhMTJhMjEyNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.nqu2M45uPAbYJ6wxqaaVfFwgoB2kWYznyYxJLaNud_E)
> 
> ![Screenshot 2023-08-14 at 09 53 33](https://private-user-images.githubusercontent.com/8404265/260379441-cfee6352-dd80-4c57-9f25-0565b5c6ae62.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE5NDk2NjYsIm5iZiI6MTcwMTk0OTM2NiwicGF0aCI6Ii84NDA0MjY1LzI2MDM3OTQ0MS1jZmVlNjM1Mi1kZDgwLTRjNTctOWYyNS0wNTY1YjVjNmFlNjIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMDdUMTE0MjQ2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODNmNDFkNDQ0MTAzMGYxOWUyNzZmMTk4NWY4MjY4YzE5MDU1ZTRlZGMzODg5ZTMwZmQxMGI1NGU2ZTNiMTg3OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.U7m6fJcWMMlJCOWIVisiszmGCCMQBR1AZA4q_HiBmio)

**Comment by Heyest** [comment]
> Furthermore, which ranks/target ranks are used to obtain results from Table 1?
> 
> E.g., For AdaLoRA with 0.32M of trainable parameters, according to Appendix C.1, I should use `r=4` and `target_rank=2`. However, scripts provided with this repository contain various ranks beyond these values.
> 
> For [CoLA](https://github.com/QingruZhang/AdaLoRA/blob/main/NLU/scripts/run_debertav3_cola.sh) and [MNLI](https://github.com/QingruZhang/AdaLoRA/blob/main/NLU/scripts/run_debertav3_mnli.sh) datasets `r=3` and `target_rank=1` For [others](https://github.com/QingruZhang/AdaLoRA/blob/main/NLU/scripts/run_debertav3_stsb.sh), `r=2` and `target_rank=1`.
> 
> Am I correctly understanding that to reproduce results from Table 1 with 0.32M parameters, I should strictly use `r=4` and `target_rank=2`?
> 
> ![Screenshot 2023-08-14 at 09 53 18](https://private-user-images.githubusercontent.com/8404265/260379384-f7944012-f528-4c56-8b2e-0a214c1f9a94.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE5NDk2NjYsIm5iZiI6MTcwMTk0OTM2NiwicGF0aCI6Ii84NDA0MjY1LzI2MDM3OTM4NC1mNzk0NDAxMi1mNTI4LTRjNTYtOGIyZS0wYTIxNGMxZjlhOTQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMDdUMTE0MjQ2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9Nzg4ZWYxMTY3OTQzNmM1ZDJlZWFjNTNhZDg3ZGFjODBmZjM5ZDkyOGE5Yzg2YjQ3YzVhNzQxMmJhMTJhMjEyNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.nqu2M45uPAbYJ6wxqaaVfFwgoB2kWYznyYxJLaNud_E)
> 
> ![Screenshot 2023-08-14 at 09 53 33](https://private-user-images.githubusercontent.com/8404265/260379441-cfee6352-dd80-4c57-9f25-0565b5c6ae62.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDE5NDk2NjYsIm5iZiI6MTcwMTk0OTM2NiwicGF0aCI6Ii84NDA0MjY1LzI2MDM3OTQ0MS1jZmVlNjM1Mi1kZDgwLTRjNTctOWYyNS0wNTY1YjVjNmFlNjIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQUlXTkpZQVg0Q1NWRUg1M0ElMkYyMDIzMTIwNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyMzEyMDdUMTE0MjQ2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODNmNDFkNDQ0MTAzMGYxOWUyNzZmMTk4NWY4MjY4YzE5MDU1ZTRlZGMzODg5ZTMwZmQxMGI1NGU2ZTNiMTg3OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.U7m6fJcWMMlJCOWIVisiszmGCCMQBR1AZA4q_HiBmio)

I've encountered the same issue. I also tried setting r=4 and target_rank=2 on the MRPC dataset, but I found that the performance even worse than the original settings in the script. Have you faced a similar issue, and how did you resolve it?
